src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -310,20 +320,49 @@ partial class BoundStateMachineScope
         protected override OperationKind StatementKind => OperationKind.None;
     }
 
-    partial class BoundLocalDeclaration : IVariableDeclarationStatement, IVariable
+    partial class BoundLocalDeclaration : IVariableDeclarationStatement
```

Comment:
BoundLocalDeclaration is used for both local declaration statement (e.g. int x; ) as well as one of the declarations of the multiple local declaration statement (e.g. declaration of x in int x, y, z;)

It is a statement in the first case, but not in the latter case (and therefore is not treated as a statement 
https://github.com/genlu/roslyn/blob/IOperationTree/src/Compilers/CSharp/Portable/BoundTree/Expression.cs#L84)